app/editor/src/features/admin/charts/ChartTemplateFormOptions.tsx

```diff
@@ -87,8 +86,20 @@ export const ChartTemplateFormOptions: React.FC<IChartTemplateFormOptionsProps>
             onChange={(e) => {
               const options = {
                 ...values.settings.options,
+                plugins: {
+                  ...values.settings.options.plugins,
+                  datalabels: {
+                    ...values.settings.options.plugins.datalabels,
+                    labels: {
+                      ...values.settings.options.plugins.datalabels.labels,
+                      title: {
+                        ...values.settings.options.plugins.datalabels.labels.title,
+                        display: e.currentTarget.checked,
+                      },
+                    },
+                  },
+                },
               };
-              _.set(options, 'plugins.datalabels.labels.title.display', e.currentTarget.checked);
```

Comment:
Lodash will not set a property in certain cases for some reason.  It's also designed to not throw any errors, so it simply ignores all attempts.